src/tests/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs

```diff
@@ -253,6 +263,249 @@ static bool CollectCrashDump(Process process, string path)
             return fSuccess && createdump.ExitCode == 0;
         }
 
+        private static List<string> knownNativeModules = new List<string>() { "libcoreclr.so", "libclrjit.so" };
+        private static string TO_BE_CONTINUE_TAG = "<TO_BE_CONTINUE>";
+        private static string SKIP_LINE_TAG = "# <SKIP_LINE>";
+
+
+        static bool RunProcess(string fileName, string arguments)
+        {
+            Process proc = new Process();
+
+            proc.StartInfo.FileName = fileName;
+            proc.StartInfo.Arguments = arguments;
+            proc.StartInfo.UseShellExecute = false;
+            proc.StartInfo.RedirectStandardOutput = true;
+            proc.StartInfo.RedirectStandardError = true;
+
+            Console.WriteLine($"Invoking: {proc.StartInfo.FileName} {proc.StartInfo.Arguments}");
+            proc.Start();
+            string stdout = proc.StandardOutput.ReadToEnd();
+            if (!string.IsNullOrEmpty(stdout) )
+            {
+                Console.WriteLine($"stdout: {stdout}");
+            }
+            string stderr = proc.StandardError.ReadToEnd();
+            if (!string.IsNullOrEmpty(stderr) )
+            {
+                Console.WriteLine($"stderr: {stderr}");
+            }
+
+            if (!proc.WaitForExit(DEFAULT_TIMEOUT_MS))
```

Comment:
I think the Process contract is that ReadToEnd() will wait for the process exit, so I'm not sure this will ever fire. You'll probably need to do async stream processing.

Reply:
Fixed.